core/src/main/scala/kafka/log/Log.scala

```diff
@@ -208,7 +208,12 @@ class Log(@volatile var dir: File,
       .format(name, segments.size(), logStartOffset, logEndOffset, time.milliseconds - startMs))
   }
 
-  private val tags = Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString)
+  private val tags = {
+    if (isFuture)
+      Map("topic" -> topicPartition.topic, "partition" -> topicPartition.partition.toString, "isFuture" -> "true")
```

Comment:
We typically use dash-separated names for kafka metrics. Should it be `is-future` then? Also, I wonder if this name will be clear to people. I'll think a bit more, don't have any concrete suggestions, right now.

Reply:
Yeah it should be is-future. I have updated the patch as suggested. Thanks!